### PR TITLE
fix: search results based on searching term

### LINKED ISSUE #370: Search by keybase name returns results as node pubkeys
It feels weird for search results to not match my input. And also it's hard to pick the right one if names start with the same string!

image

(Nodes above have names: beta-testnet-solana-com-us-west1-b-fullnode2, beta-testnet-solana-com-us-west1-b-fullnode1, beta-testnet-solana-com-bootstrap-leader)

### PR DESCRIPTION
**Pull request by manuel-calavera**
Fixes #370

### CODE REVIEW

File: src/v2/components/Search/index.jsx [hunk 1]
```diff
@@ -44,26 +44,40 @@ const Search = () => {
       return;
     }
 
-    const filteredValidators = filter(v => {
+    const filteredValidators = reduce((acc, v) => {
```
Comment (sunnygleason) [inline src/v2/components/Search/index.jsx]: This code was slightly complicated/non-obvious to me at first
with the repetition. Do you have any way to factor that out &
make the code a little easier to understand?

const condition = (x) => { A || B || C };
const results = filter(condition)(validators);

Comment (manuel-calavera) [reply src/v2/components/Search/index.jsx]: @sunnygleason it was like that before actually, look at #370 issue. these changes were made to solve it and i have to use reducers here and check for condition after each one. This way i can understand resource type.

This code will be gone after we move to search api method.

Comment (sunnygleason) [reply src/v2/components/Search/index.jsx]: @manuel-calavera ok I'll show you what I mean in a subsequent PR.

**Pull request merged**